src/Adapter/Product/Create/CombinationCreator.php

```diff
@@ -125,15 +125,19 @@ private function formatScalarValues(array $groupedAttributeIdsList): array
      *
      * @return CombinationId[]
      *
-     * @throws CombinationException
+     * @throws CombinationConstraintException
+     * @throws CoreException
      */
     private function addCombinations(Product $product, Traversable $generatedCombinations): array
     {
         $product->setAvailableDate();
+        $productId = (int) $product->id;
+        $setDefault = $this->getDefaultCombinationId($productId) ? false : true;
```

Comment:
```suggestion
        $hasDefault = (bool) $this->getDefaultCombinationId($productId);
```

Reply:
Just a suggestion to use `$hasDefault` as the variable name (so you need to call `$this->persistCombination($productId, $generatedCombination, !$hasDefault);`)

But it's just a personal opinion that it's easier to understand this way So feel free to choose what you prefer ^^